sdk/keyvault/azure-keyvault-keys/tests/test_crypto_client.py

```diff
@@ -189,6 +200,58 @@ def test_wrap_and_unwrap(self, azure_keyvault_url, **kwargs):
         result = crypto_client.unwrap_key(result.algorithm, result.encrypted_key)
         self.assertEqual(key_bytes, result.key)
 
+    @pytest.mark.skip("MHSM-only algorithms can't be tested in CI yet")
+    @KeyVaultPreparer()
+    def test_symmetric_encrypt_and_decrypt(self, azure_keyvault_url, **kwargs):
+        key_client = self.create_key_client(azure_keyvault_url)
+        key_name = self.get_resource_name("symmetric-encrypt")
+
+        imported_key = self._import_symmetric_test_key(key_client, key_name)
+        assert imported_key is not None
+        crypto_client = self.create_crypto_client(imported_key.id)
+        # Use 256-bit AES algorithms for the 256-bit key
+        symmetric_algorithms = [algo for algo in EncryptionAlgorithm if algo.startswith("A256")]
+
+        for algorithm in symmetric_algorithms:
+            if algorithm.endswith("GCM"):
+                result = crypto_client.encrypt(algorithm, self.plaintext, additional_authenticated_data=self.aad)
+                self.assertEqual(result.key_id, imported_key.id)
+                result = crypto_client.decrypt(
+                    result.algorithm,
+                    result.ciphertext,
+                    iv=result.iv,
+                    authentication_tag=result.tag,
+                    additional_authenticated_data=self.aad
+                )
+            else:
+                result = crypto_client.encrypt(
+                    algorithm, self.plaintext, iv=self.iv, additional_authenticated_data=self.aad
+                )
+                self.assertEqual(result.key_id, imported_key.id)
+                result = crypto_client.decrypt(
+                    result.algorithm, result.ciphertext, iv=self.iv, additional_authenticated_data=self.aad
+                )
+
+            self.assertEqual(result.key_id, imported_key.id)
+            self.assertEqual(algorithm, result.algorithm)
+            assert result.plaintext.startswith(self.plaintext)  # AES-CBC returns a zero-padded plaintext
```

Comment:
I would make this a special case i.e. use a strict `==` for all the other algorithms.